lib/sapience/extensions/grape/middleware/logging.rb

```diff
@@ -26,98 +25,56 @@ def initialize(app, options = {})
           def call!(env)
             @env = env
             before
-            error = catch(:error) do
-              begin
+
+            handle_raise do
+              @error = catch(:error) do
                 @app_response = @app.call(@env)
-              rescue StandardError => e
-                after_exception(e)
-                raise e
+                nil # no error thrown
+              end
+              if @error
+                @status = @error[:status]
+                throw(:error, @error)
               end
-              nil
-            end
-            if error
-              after_failure(error)
-              throw(:error, error)
-            else
-              @status, = *@app_response
-              after
             end
             @app_response
           end
 
-          def before
-            reset_db_runtime
-            start_time
-          end
-
-          def after
-            stop_time
-            @logger.info(parameters)
-          end
-
-          def after_exception(exc) # rubocop:disable Lint/UnusedMethodArgument
-            @status = 500
-            after
-          end
-
-          def after_failure(error)
-            @status = error[:status]
-            after
+          def handle_raise
+            begin
+              yield
+            rescue StandardError => e
+              if e.class.name =~ %r{NotFound}
```

Comment:
The regex matching on class name will surely cause pain in future refactorings. IMO it should try to match the parent class type of a type hierarchy.

Reply:
I guess you didnt watch this vid? - https://www.youtube.com/watch?v=IRTfhkiAqPw
How can we ask used of the gem to place all 404 type errors in a hierarchy. 
What hierarchy does ActiveRecord::RecordNotFound fall into?